gprofiler/profilers/java.py

```diff
@@ -257,14 +257,45 @@ def is_java_basename(process: Process) -> bool:
 _JAVA_VERSION_CACHE_MAX = 1024
 
 
+def _get_process_ns_java_path(process: Process) -> Optional[str]:
+    """
+    Look up path to java executable installed together with this process' libjvm.
+    """
+    java_path: Optional[str] = None
+    nspid = get_process_nspid(process.pid)
+    if is_java_basename(process):
+        java_path = f"/proc/{nspid}/exe"  # it's a symlink and will be resolveable under process' mnt ns
+    else:
+        libjvm_path: Optional[str] = None
+        for m in process.memory_maps():
+            if re.match(DETECTED_JAVA_PROCESSES_REGEX, m.path):
+                libjvm_path = m.path
+                break
+        if libjvm_path is None:
+            return None
+        libjvm_dir = os.path.dirname(libjvm_path)
+        # support two java layouts - it's either lib/server/../../bin/java or lib/{arch}/server/../../../bin/java:
+        java_candidate_paths = [
+            Path(libjvm_dir, "../../bin/java").resolve(),
+            Path(libjvm_dir, "../../../bin/java").resolve(),
+        ]
+        for p in java_candidate_paths:
+            proc_relative_path = Path(f"/proc/{process.pid}/root", p.relative_to("/"))
+            if proc_relative_path.exists():
+                if os.access(proc_relative_path, os.X_OK):
+                    java_path = os.path.join(f"/proc/{nspid}/root", p)
```

Comment:
This doesn't make much sense to me. This path will be accessed inside the target process mount ns - so there's no reason to use `/proc/nspid/root`, as `/proc/nspid/root` is logically equivalent to `/`.